dace/transformation/auto/auto_optimize.py

```diff
@@ -259,23 +316,29 @@ def move_small_arrays_to_stack(sdfg: SDFG) -> None:
                 and array.lifetime == dtypes.AllocationLifetime.Scope):
             if not symbolic.issymbolic(array.total_size, sd.constants):
                 eval_size = symbolic.evaluate(array.total_size, sd.constants)
-                if eval_size <= tile_size:
-                    array.storage = dtypes.StorageType.Register
-                    converted += 1
+                try:
+                    if eval_size <= tile_size:
+                        array.storage = dtypes.StorageType.Register
+                        converted += 1
+                except TypeError:
+                    # cannot determine truth value of relational
+                    pass
 
     if config.Config.get_bool('debugprint') and converted > 0:
         print(f'Statically allocating {converted} transient arrays')
 
 
 def set_fast_implementations(sdfg: SDFG,
                              device: dtypes.DeviceType,
-                             blocklist: List[str] = None):
+                             blocklist: List[str] = [],
```

Comment:
do not put mutable objects on default arguments (they keep state between function calls). Use `= None` instead and `blocklist = blocklist or []` within the function